iup/__init__.py

```diff
@@ -0,0 +1,739 @@
+
+        Details
+        Assume overwinter seasons, e.g. 2023-10-07 and 2024-04-18 are both in "2023/24"
+        """
+        year1 = (
+            date_col.dt.year() + pl.when(date_col.dt.month() < 7).then(-1).otherwise(0)
+        ).cast(pl.Utf8)
+        year2 = (
+            date_col.dt.year() + pl.when(date_col.dt.month() < 7).then(0).otherwise(1)
+        ).cast(pl.Utf8)
+        season = pl.concat_str([year1, year2], separator="/")
+
+        return season
+
+    @staticmethod
+    def date_to_interval(date_col: pl.Expr) -> pl.Expr:
+        """
+        Extract a time interval column from a date column, as polars expressions.
+
+        Parameters
+        date_col: pl.Expr
+            column of dates
+
+        Returns
+        pl.Expr
+            column of the number of days between each date and the previous
+
+        Details
+        Date column should be chronologically sorted in advance.
+        Time difference is always in days.
+        """
+        interval = date_col.diff().dt.total_days().cast(pl.Float64)
+
+        return interval
+
+    @staticmethod
+    def date_to_elapsed(date_col: pl.Expr) -> pl.Expr:
+        """
+        Extract a time elapsed column from a date column, as polars expressions.
+
+        Parameters
+        date_col: pl.Expr
+            column of dates
+
+        Returns
+        pl.Expr
+            column of the number of days elapsed since the first date
+
+        Details
+        Date column should be chronologically sorted in advance.
+        Time difference is always in days.
+        """
+        elapsed = (date_col - date_col.first()).dt.total_days().cast(pl.Float64)
+
+        return elapsed
+
+    @staticmethod
+    def split_train_test(uptake_data_list, start_date: dt.date, side: str):
+        """
+        Concatenate UptakeData objects and split into training and test data.
+
+        Parameters
+        uptake_data_list: List[UptakeData]
+            cumulative or incident uptake data objects, often from different seasons
+        start_date: dt.date
+            the first date for which projections should be made
+        side: str
+            whether the "train" or "test" portion of the data is desired
+
+        Returns
+        UptakeData
+            cumulative or uptake data object of the training or test portion
+
+        Details
+        Training data are before the start date; test data are on or after.
+        """
+        orig_class = type(uptake_data_list[0])
+        if side == "train":
+            out = (
+                pl.concat(uptake_data_list)
+                .sort("date")
+                .filter(pl.col("date") < start_date)
+            )
+        if side == "test":
+            out = (
+                pl.concat(uptake_data_list)
+                .sort("date")
+                .filter(pl.col("date") >= start_date)
+            )
+
+        out = orig_class(out)
+
+        return out
+
+
+class IncidentUptakeData(UptakeData):
```

Comment:
I found in the fall virus work that it was good to have columns like `inc_uptake` or `cum_uptake`, to be absolutely crystal clear about what the "estimate" was

something for the backlog

Reply:
Issue #44